lib/Test/Script.pm

```diff
@@ -580,6 +597,21 @@ sub program_runs {
   goto &_run;
 }
 
+=head2 program_fails
+
+L</program_runs> may be invoked as L</program_fails>. Unlike L</script_fails> which defaults to 255 (the exit value for C<die>), L</program_fails> needs to know the expected exit value, so exit becomes a required option. 
+
+=cut
+
+sub program_fails {
+  my $cmd    = _script(shift);
+  my ( $opt, $testname ) = @_;
+  die "exit is a mandatory option for program_fails"
+    unless eval{ defined $opt->{exit} };
+  $testname = 'program_fails' unless defined $testname;
+  program_runs( $cmd, $opt, $testname );
```

Comment:
Will need to grab and release a context here too in order to get the diagnostic to point to the right place.